### PR TITLE
refactor(Rating): match Figma and utilize Tailwind

### PR DESCRIPTION
**Pull request by justinhough**
**Related Issue(s):** #2810 (originally part of #2041)

## Summary

Updated heights for S `24px`, M `32px`, and L `44px` to work without chip active

### CODE REVIEW

**Review by julio8a — APPROVED** [review]
Visually, looks good! I do see the 24, 36, and 44 when inspecting without label. 👍

File: src/components/calcite-rating/calcite-rating.scss [hunk 1]
```diff
@@ -12,17 +12,25 @@
 }
 
 :host([scale="s"]) {
+  @apply flex h-6;
   --calcite-rating-spacing-unit: theme("spacing.1");
 }
 
 :host([scale="m"]) {
+  @apply flex h-8;
```
Comment (caripizza) [inline src/components/calcite-rating/calcite-rating.scss:20]: remove `flex` here too

File: src/components/calcite-rating/calcite-rating.scss [hunk 2]
```diff
@@ -12,17 +12,25 @@
...
   --calcite-rating-spacing-unit: theme("spacing.2");
 }
 
 :host([scale="l"]) {
+  @apply flex h-11;
```
Comment (caripizza) [inline src/components/calcite-rating/calcite-rating.scss:25]: remove `flex` here too

File: src/components/calcite-rating/calcite-rating.scss [hunk 3]
```diff
@@ -12,17 +12,25 @@
...
   --calcite-rating-spacing-unit: theme("spacing.3");
 }
 
+:host .fieldset,
+:host label {
+  @apply flex;
```
Comment (caripizza) [inline src/components/calcite-rating/calcite-rating.scss:31]: Remove this chunk of code too

**Review by caripizza — CHANGES_REQUESTED** [review]
Looking good @justinhough! Left a couple comments on flex additions that can be removed. Instead, we can target the existing [`.fieldset`](https://github.com/Esri/calcite-components/blob/master/src/components/calcite-rating/calcite-rating.scss#L35) and [`.star`](https://github.com/Esri/calcite-components/blob/master/src/components/calcite-rating/calcite-rating.scss#L50) selectors in the stylesheet to change their displays to `inline-flex` or `flex`.

**Comment by julio8a** [comment]
@caripizza, can you give this one another review?

**Review by caripizza — APPROVED** [review]
👍  Can you remove the extra line of whitespace and re-merge in master branch?

**Comment by justinhough** [comment]
@caripizza Both have been completed.

**Comment by julio8a** [comment]
This PR has unrelated test failure, going to move forward with merging.

**Pull request merged**